File: src/api/ApiResponse.ts
import ProblemDetails from "../models/ProblemDetails";

export default interface ApiResponse<T = unknown> {
  success: boolean;
  data?: T;
  errorMessage?: string;
  statusCode?: number;
  /** Structured error details supplied by the API, when the failed response uses that schema. */
  problemDetails?: ProblemDetails;
}

File: src-dotnet/api/ApiResponse.cs
namespace MedrunnerApiClient.Api
{
    /// <summary>
    /// Represents a generic API response.
    /// </summary>
    public class ApiResponse<T>
    {
        public bool Success { get; set; }
        public T? Data { get; set; }
        public string? ErrorMessage { get; set; }
        public int? StatusCode { get; set; }
    }
}
